openjdk/mmtkBarrierSetC1.cpp

```diff
@@ -4,18 +4,58 @@
 #include "mmtkBarrierSetC1.hpp"
 
 void MMTkBarrierSetC1::generate_c1_runtime_stubs(BufferBlob* buffer_blob) {
-  class MMTkBarrierCodeGenClosure : public StubAssemblerCodeGenClosure {
+  class MMTkPreBarrierCodeGenClosure : public StubAssemblerCodeGenClosure {
+
     virtual OopMapSet* generate_code(StubAssembler* sasm) override {
       MMTkBarrierSetAssembler* bs = (MMTkBarrierSetAssembler*) BarrierSet::barrier_set()->barrier_set_assembler();
-      bs->generate_c1_write_barrier_runtime_stub(sasm);
+      bs->generate_c1_pre_write_barrier_runtime_stub(sasm);
       return NULL;
     }
+  public:
+    MMTkPreBarrierCodeGenClosure() {}
   };
-  MMTkBarrierCodeGenClosure write_code_gen_cl;
-  _write_barrier_c1_runtime_code_blob = Runtime1::generate_blob(buffer_blob, -1, "write_code_gen_cl", false, &write_code_gen_cl);
+
+  class MMTkPostBarrierCodeGenClosure : public StubAssemblerCodeGenClosure {
+    virtual OopMapSet* generate_code(StubAssembler* sasm) override {
+      MMTkBarrierSetAssembler* bs = (MMTkBarrierSetAssembler*) BarrierSet::barrier_set()->barrier_set_assembler();
+      bs->generate_c1_post_write_barrier_runtime_stub(sasm);
+      return NULL;
+    }
+  public:
+    MMTkPostBarrierCodeGenClosure() {}
+  };
+
+  MMTkPreBarrierCodeGenClosure pre_write_code_gen_cl;
+  _pre_barrier_c1_runtime_code_blob = Runtime1::generate_blob(buffer_blob, -1, "mmtk_pre_write_code_gen_cl", false, &pre_write_code_gen_cl);
+  MMTkPostBarrierCodeGenClosure post_write_code_gen_cl;
+  _post_barrier_c1_runtime_code_blob = Runtime1::generate_blob(buffer_blob, -1, "mmtk_post_write_code_gen_cl", false, &post_write_code_gen_cl);
+  // MMTkBarrierCodeGenClosure write_code_gen_cl_patch_fix(true);
```

Comment:
This was in the lxr branch and after I discussed with wenyu, we believe it is redundant. The code patching has already been dealt with, no need to do any special things here